mvc/models/model-binding.rst

```diff
@@ -1,8 +1,105 @@
+
+Once model binding is complete, validation occurs. The default model binders are quite smart, and work great for the vast majority of development scenarios. However, ASP.NET MVC is extensible so if you have unique needs you can customize the built in model binders as you wish.  
+
+Customize default model binding behavior with attributes 
+--------------------------------------------------------
+ASP.NET MVC contains several attributes that you can use to direct its default model binding behavior to a different source. For example, you can specify whether binding is required for a field, or if it should never happen at all by using the ``[BindRequired]`` or ``[BindNever]`` attributes. Alternatively, you can override the default value, and specify the model binder's data source, which is what the ``[From*]`` attributes do, such as the ``[FromForm]`` or ``[FromQuery]``. While the topic of model binding extensibility in depth is outside of the scope of this doc, here are a few customizations that you should be aware of:
+
+- ``[BindRequired]``: This attribute forces a Model state error if binding cannot occur.
+- ``[BindNever]``: Tells the model binder to never bind to this parameter.
+- ``[FromHeader]``, ``[FromQuery]``, ``[FromRoute]``, ``[FromForm]``: Use these to specify the exact binding source you want to apply.
+- ``[FromServices]``: Only supported on action parameters. 
+- ``[FromBody]``: Instructs the model binder to use the JSON media type formatter rather than regular model binding.
+- ``[ModelBinder]``: You reject the current model binder and substitute it with your own.
+
+A good example of binding attribute usage in the wild is when you see the ``[FromBody]`` attribute in action methods of API Controllers. While there are only a few differences between standard MVC and API Controllers, one such difference is that API Controllers attempt to retrieve data from the request URI in PUT and POST messages, rather than from the body by default. This means that when a page sends data via an Ajax call, Web API actions still look in the header, but applying the ``[FromBody]`` attribute will direct the model binder to the correct data source, as shown in the snippet below:
+
+.. literalinclude:: model-binding/sample/src/MVCMovie/Controllers/MoviesAPIController.cs
+   :language: c#
+   :lines: 47-49
+
+Binding attributes are very helpful tools when you need to nudge the default behaviors of model binders.
+
+Binding formatted data from the request body
+--------------------------------------------
+Formatting data is an important behavior of controllers, and it influences model binding behavior as well. A formatter is an object that accepts data and serializes or deserializes it into some shape. JSON is the most popular format, as it's the smallest and most lightweight way to structure textual data. It's also the default format for API controllers. Below is a list of several built-in formatters:  
```

Comment:
Why is the model binding topic talking about formatters? Should there instead be a paragraph here that says something like "if your scenario is XYZ then instead of model binding there is an alternative called formatters, click here to read more"?